src/main/java/uk/gov/ons/census/casesvc/logging/EventLogger.java

```diff
@@ -71,6 +73,30 @@ public void logRefusalEvent(
         eventMetaDataDateTime);
   }
 
+  public void logFulfilmentRequestedEvent(
+      Case caze,
+      UUID caseId,
+      String eventDescription,
+      EventType eventType,
+      FulfilmentRequestDTO payload,
+      EventDTO event) {
+
+    Event loggedEvent = new Event();
+    loggedEvent.setCaze(caze);
+    loggedEvent.setId(UUID.randomUUID());
+    loggedEvent.setCaseId(caseId);
+    loggedEvent.setEventDate(OffsetDateTime.now());
+    loggedEvent.setEventDescription(eventDescription);
+    loggedEvent.setEventType(eventType);
+    loggedEvent.setEventPayload(convertObjectToJson(payload));
+    loggedEvent.setEventChannel(event.getChannel());
+    loggedEvent.setEventSource(event.getSource());
+    loggedEvent.setEventTransactionId(UUID.randomUUID());
```

Comment:
shouldn't transaction id be from event?

Reply:
Changes made in my last commit to address this, let me know what you think

Reply:
that's the best you can do with confirmation from dave